src/input/pointer/grab.rs

```diff
@@ -124,6 +132,16 @@ impl<D: SeatHandler + 'static> PointerGrab<D> for DefaultGrab {
         handle.motion(data, focus, event);
     }
 
+    fn relative_motion(
```

Comment:
The relative-pointer protocol states:
> A wp_relative_pointer object shares the same focus as wl_pointer objects of the same seat and will only emit events when it has focus.

Which means we could maybe skip piping this through all of the grab-handling and just pass it to a focused object, if any was determined by the last `motion`-call?

Alternatively (and I think I even like that approach better), we could just extend the `MotionEvent` to contain an optional relative-field?

Reply:
> Alternatively (and I think I even like that approach better), we could just extend the MotionEvent to contain an optional relative-field?

Hm, I think that could work, and would be cleaner here. Currently `MotionEvent` is also passed to `enter()` but I guess it could just have `None` for relative if it's optional.

Reply:
https://wayland.app/protocols/relative-pointer-unstable-v1

> Relative motions are not coupled to wl_pointer.motion events, and can be sent in combination with such events, but also independently. There may also be scenarios where wl_pointer.motion is sent, but there is no relative motion. The order of an absolute and relative motion event originating from the same physical motion is not guaranteed.

This doesn't really clarify when/if it makes sense for the compositor to produce one but not the other.

I guess if the mouse is moved at the edge of the screen, it may produce a relative motion, but still remain at the same absolute position, and not emit an event. Or https://wayland.app/protocols/pointer-constraints-unstable-v1#zwp_pointer_constraints_v1:request:lock_pointer may cause it to not have any absolute motion, but only relative motion events.

Reply:
Yep if pointer is constrained (eg. like in a game) there will be no motion events, only relative ones, so I suppose keeping it separate makes sense here.